docs/utils/getPackages.ts

```diff
@@ -0,0 +1,123 @@
+  );
+
+const localReadmeFromSlug = (slug: string): string =>
+  fs.readFileSync(`${localPackageDirFromSlug(slug)}/README.md`, { encoding: 'utf-8' });
+
+const localPackageDataFromSlug = (slug: string): PackageData => ({
+  slug,
+  metadata: localPackageMetadataFromSlug(slug),
+  readme: localReadmeFromSlug(slug),
+});
+
+const getAllLocalPackageData = (): PackageData[] =>
+  getLocalPackageSlugs().map(localPackageDataFromSlug);
+
+const festivePossumPackagesPath =
+  'https://raw.githubusercontent.com/kofno/festive-possum/main/packages' as const;
+
+type GithubFestivePossumPath = `${typeof festivePossumPackagesPath}/${string}`;
+
+interface FestivePossumPackagePaths {
+  slug: string;
+  readme: GithubFestivePossumPath;
+  packageJson: GithubFestivePossumPath;
+}
+
+const festivePossumPackage = (slug: string, readmeName: string): FestivePossumPackagePaths => ({
+  slug,
+  readme: `${festivePossumPackagesPath}/${slug}/${readmeName}`,
+  packageJson: `${festivePossumPackagesPath}/${slug}/package.json`,
+});
+
+const festivePossumReadmes = {
+  ajaxian: 'readme.md',
+  ajaxios: 'readme.md',
+  cute: 'README.md',
+  gaia: 'readme.md',
+  jsonous: 'readme.md',
+  maybeasy: 'readme.md',
+  'nonempty-list': 'README.md',
+  piper: 'README.md',
+  resulty: 'readme.md',
+  'stack-ts': 'README.md',
+  taskarian: 'readme.md',
+};
+
+const festivePossumPaths: FestivePossumPackagePaths[] = Object.entries(festivePossumReadmes).map(
+  ([slug, readme]) => festivePossumPackage(slug, readme)
+);
+
+const festivePossumPackageMetadata = (
+  paths: FestivePossumPackagePaths
+): Task<HttpError, PackageMetadata> =>
+  toHttpTask(get(paths.packageJson).withDecoder(packageMetadataDecoder));
+
+const festivePossumPackageReadmeContent = (
+  paths: FestivePossumPackagePaths
+): Task<HttpError, string> => toHttpTask(get(paths.readme).withDecoder(string));
+
+const festivePossumPackageData = (paths: FestivePossumPackagePaths): Task<HttpError, PackageData> =>
+  Task.succeed<HttpError, { slug: string }>({ slug: paths.slug })
+    .assign('metadata', festivePossumPackageMetadata(paths))
+    .assign('readme', festivePossumPackageReadmeContent(paths));
+
+const getAllFestivePossumPackageData = (): Task<HttpError, Array<PackageData>> =>
+  festivePossumPaths.reduce<Task<HttpError, Array<PackageData>>>(
+    (t, paths) => t.andThen((a) => festivePossumPackageData(paths).map((d) => [...a, d])),
+    Task.succeed([])
+  );
+
+export const getCombinedPackageData = (): Task<HttpError, Array<PackageData>> =>
+  getAllFestivePossumPackageData().map((a) => [...a, ...getAllLocalPackageData()]);
+
+export const getPackageDataFromSomeSlug = (slug: string): Task<HttpError, PackageData> => {
+  if (Object.hasOwn(festivePossumReadmes, slug)) {
+    return festivePossumPackageData(
+      festivePossumPackage(slug, festivePossumReadmes[slug as keyof typeof festivePossumReadmes])
+    );
+  } else {
+    return Task.succeed<HttpError, PackageData>(localPackageDataFromSlug(slug));
+  }
+};
```

Comment:
To remove type assertion from this code,

```ts
const festivePossumReadmes = {
// ...
} as const;

type FestivePossumPackageName = keyof typeof festivePossumReadmes;
```

```ts
const isFestivePossumPackageName = (slug: string): slug is FestivePossumPackageName =>
  Object.hasOwn(festivePossumReadmes, slug);

export const getPackageDataFromSomeSlug = (slug: string): Task<HttpError, PackageData> => {
  if (isFestivePossumPackageName(slug)) {
    return festivePossumPackageData(festivePossumPackage(slug, festivePossumReadmes[slug]));
  } else {
    return Task.succeed<HttpError, PackageData>(localPackageDataFromSlug(slug));
  }
};
```

Reply:
I ended up making `festivePossumReadmes` a `Map` type instead. It was a bit easier to work with.